%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% Stylish Article
% LaTeX Template
% Version 1.0 (31/1/13)
%
% This template has been downloaded from:
% http://www.LaTeXTemplates.com
%
% Original author:
% Mathias Legrand (legrand.mathias@gmail.com)
%
% License:
% CC BY-NC-SA 3.0 (http://creativecommons.org/licenses/by-nc-sa/3.0/)
%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%----------------------------------------------------------------------------------------
%    PACKAGES AND OTHER DOCUMENT CONFIGURATIONS
%----------------------------------------------------------------------------------------

\documentclass[fleqn,10pt]{SelfArx} % Document font size and equations flushed left

\setlength{\columnsep}{0.55cm} % Distance between the two columns of text
\setlength{\fboxrule}{0.75pt} % Width of the border around the abstract

\definecolor{color1}{RGB}{139,0,0}% Color of the article title and sections
\definecolor{color2}{RGB}{0,205, 205}% Color of the boxes behind the abstract and headings

\newlength{\tocsep} 
\setlength\tocsep{1.5pc} % Sets the indentation of the sections in the table of contents
\setcounter{tocdepth}{3} % Show only three levels in the table of contents section: sections, subsections and subsubsections

\usepackage{lipsum} % Required to insert dummy text
\usepackage[english]{babel} %more dumb text
\usepackage{blindtext}
\usepackage{pdfpages}

%----------------------------------------------------------------------------------------
%    ARTICLE INFORMATION
%----------------------------------------------------------------------------------------

\JournalInfo{Science Translational Medicine (IF = 10.76) \\ Cancer Research (8.650) \\ 
The Journal of Pathology, (7.58)\\ Neuro-Oncology (6.180)\\
Molecular Cancer (5.13)\\ European Journal of Cancer (5.06)\\

or similar \\TRANSLATIONAL RESEARCH journal} % Journal information

\Archive{Original article} % Additional notes (e.g. copyright, DOI, review/research article)

\PaperTitle{CD133 and CD44 expressing subpopulations of Glioblastoma Multiforme predominate in different tumour locations and have distinct transcriptonal profiles} % Article title

\Authors{Brown, Daniel V\textsuperscript{1}*, Ng, Wayne\textsuperscript{2}, Amiridis, Stephanie\textsuperscript{2}, 
Kountouri, Nicole\textsuperscript{2}, Daniel, Paul M\textsuperscript{1}, Filiz, Gulay\textsuperscript{1}, Christie, Michael\textsuperscript{3},
Hollande, Fred\textsuperscript{4}, Drummond, Kate\textsuperscript{2},
Lonie, Andrew\textsuperscript{5}, Morokoff, Andrew\textsuperscript{2}, Mantamadiotis, Theo\textsuperscript{1}} % Authors

\affiliation{\textsuperscript{1}\textit{Cancer Signalling laboratory, Department of Pathology, University of Melbourne, Melbourne, Australia}} % Author affiliation
\affiliation{\textsuperscript{2}\textit{Department of Surgery, Royal Melbourne Hospital, Melbourne, Australia}} % Author affiliation
\affiliation{\textsuperscript{1}\textit{Cancer Diagnostics laboratory, Department of Pathology, University of Melbourne, Melbourne, Australia}}
\affiliation{\textsuperscript{4}\textit{Molecular Mechanisms of Tumour Progression, Department of Pathology, Melbourne, Australia}}
\affiliation{\textsuperscript{5}\textit{Victorian Life Sciences Computational Initiative, Melbourne, Australia}} % Author affiliation
\affiliation{*\textbf{Corresponding author}: dvbrown@student.unimelb.edu.au} % Corresponding author

\Keywords{Glioblastoma --- Cancer Stem Cells --- Invasion --- Xenograft --- Orthotopic model} % Keywords - if you don't want any simply remove all the text between the curly brackets
\newcommand{\keywordname}{Keywords} % Defines the keywords heading name

%----------------------------------------------------------------------------------------
%	ABSTRACT
%----------------------------------------------------------------------------------------

\Abstract{
Glioblastoma multiforme (GBM) is an aggressive cancer of the brain with a median survival of 14 months. This has been ascribed to the the difficulty of surgical resection due to the invasive nature of GBM. GBM has been shown to be maintained by a subpopulation of cells termed glioma-initiating cells (GICs). It may be these cells that are responsible for invasive potential of high grade gliomas.\\

An orthotopic model of GIC driven GBM was used to examine the transcriptome of cells sorted by flow cytometry (FACS) for the putative stem cell markers CD133 and CD44.
The xenograft derived cells at the core of the tumour and the tumour margins were also compared and contrasted with the sorted cells.
All 4 permutations of the CSC cells sorted by FACS were characterised by RNA-seq. Differential gene expression testing, exon usage and clustering were performed to resolve which transcriptional signature the CSC markers define \emph{in vivo} and were related to the tumour core and margin signature by means of clustering and associated machine learning algorithms.
There is evidence that CD133 and CD44 expressing cells are genetically distinct so it may be worth doing the TruSeq cancer panel.\\

My hypothesis is that the tumour margins will express CD44, the peri-hypoxic core will express CD133 and neither will be a marker of stemness. CD44 will mark migration/ invasion and CD133 will mark proliferating/ transit amplifying cells. Potentially the double positive cells will be cancer stem cells.\\

This paper reveals novel insights into the signalling pathways that drive the tumour margin and tumour core and the difference between the phenotype of CD44 and CD133 expressing cells \emph{in vivo}. If I can get some primary tumour slides and stain them for CD133 and CD44 it will push this article into the next echelon of journals.
}

\begin{document}
<<setup, include=FALSE, cache=FALSE>>=
library(knitr)
options(replace.assign=TRUE,width=50)
opts_chunk$set(fig.path='figure/graphics-', cache.path='cache/graphics-', fig.align='center', fig.width=5, fig.height=5, fig.show='hold', cache=TRUE, par=TRUE)
knit_hooks$set(par=function(before, options, envir){
if (before && options$fig.show!='none') par(mar=c(4,4,.1,.1),cex.lab=.95,cex.axis=.9,mgp=c(2,.7,0),tcl=-.3)
}, crop=hook_pdfcrop)
@
%\SweaveOpts{concordance=TRUE}
%This says that figures should be resized so that they fit the full width of the text.

\flushbottom % Makes all text pages the same height

\maketitle % Print the title and abstract box

\tableofcontents % Print the contents section

% \thispagestyle{empty} % Removes page numbering from the first page

%----------------------------------------------------------------------------------------
%    INTRODUCTION
%----------------------------------------------------------------------------------------
\section*{Introduction} % The \section*{} command stops section numbering

\addcontentsline{toc}{section}{\hspace*{-\tocsep}Introduction} % Adds this section to the table of contents with negative horizontal space equal to the indent for the numbered sections
  \subsection*{Some stats on prevalence and mortality}
  
Glioblastoma multiforme (GBM) is an aggressive, heterogeneous tumour of the central nervous system. These diverse morphological features have made diagnosis of GBM difficult (1). Molecular profiling of GBM tumours has revealed that similar to other tumours such as breast and ovarian cancer (2,3), GBM is not a single disease but consists of several distinct malignancies. Although GBM is a relatively rare type of cancer it has a 5 year survival of less than 5\% rendering it one of the most lethal types of tumours. Age is the most important risk factor for GBM, with incidence showing a 4th power relationship with age, indicative of a gradual accumulation of oncogenic mutations over time (7).

\subsection*{Some info on invasion and its link to cancer stem cells}  
    \paragraph {Link EMT to CSC and associated markers}
\blindtext

\subsection*{The limitations of in vitro and subcutaneous models}  
    \paragraph {Comment on how my in vitro paper is garbage}
\blindtext
    
%----------------------------------------------------------------------------------------
%    METHODS
%----------------------------------------------------------------------------------------
\section*{Methods} % The \section*{} command stops section numbering

\addcontentsline{toc}{section}{\hspace*{-\tocsep}Methods} % Adds this section to the table of contents with negative horizontal space equal to the indent for the numbered sections
  \subsection*{Cell Culture}  
    \paragraph {Culture of Glioma stem cells} a\\
\blindtext

  \subsection*{Orthotopic model of CSC driven GBM}  
    \paragraph {Cite Wayne's paper} a\\
\blindtext

  \subsection*{Flow cytometry}  
    \paragraph {The important part of gating} a\\
\blindtext 
    
  \subsection*{Immunohistochemistry}  
  \paragraph {How I solved the difficult problem of optimising CD133 \& CD44 IHC} a\\

  \subsection*{Library preparation and RNA sequencing}  
     \paragraph {RNA-seq libraries} a\\
\blindtext
    
  \subsection*{Bioinformatic analysis} 
    \paragraph {Preprocessing} a\\
    \paragraph {Differential expression testing} a\\
    \paragraph {Subtype classification} a\\
    \paragraph {Gene signature and survival analysis} a\\
\blindtext


%----------------------------------------------------------------------------------------
%    RESULTS
%----------------------------------------------------------------------------------------
\section*{RESULTS} % The \section*{} command stops section numbering

\addcontentsline{toc}{section}{\hspace*{-\tocsep}Results} % Adds this section to the table of contents with negative horizontal space equal to the indent for the numbered sections

  \subsection*{Validation of the orthotopic model}
    \paragraph {Survival curves on the 5 clones that Wayne, Steph and myself prepared} a\\
    \paragraph {Flow cytometry comparisons of CD133 and CD44} \blindtext \\
    \paragraph {IHC of stem, differentiation markers at tumour core and margin} a\\
\blindtext

\begin{figure}[ht]
   \centering 
% Change this plot when you change the data source   
    \includegraphics[page=1, width=.55\linewidth]{/Users/d.brown6/Documents/facsData/flowJo/analysis/140513_tripleStain/controls.pdf}\\
    \includegraphics[page=1, width=.55\linewidth]{/Users/d.brown6/Documents/facsData/flowJo/analysis/140513_tripleStain/samples.pdf}
    \caption{ {How the GSCs were FACS sorted on CD133 and CD44 and the basis of the comparisons I use} }
    \label{fig:FACS}
\end{figure}

%------------------------------------------------------------------------------------------------------------------------

  \subsection *{Comparing the transcriptome of the CD133, CD44 populations}
    \paragraph {RNA-seq of the 4 permuations of the FACS sorted samples and the tumour margin and core.}
\blindtext

\begin{figure}[ht]
   \centering 
% Change this plot when you change the data source  
    \includegraphics[page=1, width=.45\linewidth]{/Users/d.brown6/Documents/RNAdata/danBatch1/bowtieGem/revHTSeq/GLMedgeR/140203_facsBatch/140203_summaries.pdf}
    \includegraphics[page=1, width=.45\linewidth]{/Users/d.brown6/Documents/RNAdata/danBatch1/bowtieGem/revHTSeq/networks/131104_imageLegendBold.pdf}
    \includegraphics[page=1, width=.45\linewidth]{/Users/d.brown6/Documents/RNAdata/danBatch1/GSEA/original/131025_nonSignificantNatureGenes.pdf}
    
    \caption{{The suite of genes that are differentially expressed between CD133 and CD44} }
    \label{fig:marginCore}
\end{figure}


%------------------------------------------------------------------------------------------------------------------------

  \subsection*{Comparison of the transcriptome of the tumour margin and core}  
    \paragraph {Dependent on getting a resonable amount of material to RNA-seq the tumour margin that is free of mouse tissue contamination.}
\blindtext

\begin{figure}[ht]
   \centering 
% Change this plot when you change the data source  
    \includegraphics[page=1, width=.45\linewidth]{/Users/d.brown6/Documents/RNAdata/danBatch1/bowtieGem/revHTSeq/GLMedgeR/140203_facsBatch/140203_summaries.pdf}
    \includegraphics[page=1, width=.45\linewidth]{/Users/d.brown6/Documents/RNAdata/danBatch1/bowtieGem/revHTSeq/networks/131104_imageLegendBold.pdf}
    \includegraphics[page=1, width=.45\linewidth]{/Users/d.brown6/Documents/RNAdata/danBatch1/GSEA/original/131025_nonSignificantNatureGenes.pdf}
    
    \caption{{The suite of genes that are differentially expressed between tumour core and margin} }
    \label{fig:CD133_CD44}
 \end{figure}

%------------------------------------------------------------------------------------------------------------------------

  \subsection*{Characteristics of the mutations before and after engraftment in mouse}  
    \paragraph {If the lab wants to spend the money on the TruSeq Cancer Amplicon}
\blindtext

%------------------------------------------------------------------------------------------------------------------------

  \subsection*{Pattern of CD133 and CD44 expression in human samples}  
    \paragraph {Provided I can get some sections and also if I can get the antibodies to work well}
\blindtext

\begin{figure}[ht]
   \centering 
% Change this plot when you change the data source  
    \includegraphics[page=1, width=.45\linewidth]{/Users/d.brown6/Documents/Cell_biology/microscopy/ELDA/140226_2ndyearReviewSummary/140226_2ndYearSummary.pdf}
    \includegraphics[page=1, width=.45\linewidth]{/Users/d.brown6/Documents/Cell_biology/microscopy/ELDA/140226_2ndyearReviewSummary/140227_percentPrimaryRecurrent.pdf}
    \caption{{Pattern of CD133 and CD44 in humans and associated quantification} }
    \label{fig:IHC}
 \end{figure}

%------------------------------------------------------------------------------------------------------------------------

  \subsection*{Gene signature and survival analysis}  
    \paragraph {Use the TCGA data to examine how relevant the signatures I have discovered are to predict survival}
\blindtext

\begin{figure}[ht]
   \centering 
% Change this plot when you change the data source   
    \includegraphics[page=1, width=.75\linewidth]{/Users/d.brown6/Documents/stemCellSig/130117_signature/130123_collatedFigure.pdf}
    \caption{{A dummy figure illustrating survival analysis}  }
    \label{fig:Survival}
 \end{figure}

%------------------------------------------------------------------------------------------------------------------------



%----------------------------------------------------------------------------------------
%    DISCUSSION
%----------------------------------------------------------------------------------------
\section*{Discussion} % The \section*{} command stops section numbering

\addcontentsline{toc}{section}{\hspace*{-\tocsep}Discussion} % Adds this section to the table of contents with negative horizontal space equal to the indent for the numbered sections
  \subsection*{I will wildy speculate on my results here}  
    \paragraph {Say something clever here about the translational potential of my work}
\lipsum


%----------------------------------------------------------------------------------------
%    ACKNOWLEDGEMENTS
%----------------------------------------------------------------------------------------
\section*{Acknowledgements} % The \section*{} command stops section numbering

\addcontentsline{toc}{section}{\hspace*{-\tocsep}Acknowledgements} % Adds this section to the table of contents with negative horizontal space equal to the indent for the numbered sections
    \paragraph {Shout out to}
Many thanks to Gulay Filiz for ordering my reagents, Nicole Kountouri and Wayne Ng for discussions about stem cells. Apprieciation to House of cards, Dr Daxx and Haymarket for caffinated sustenance.

%----------------------------------------------------------------------------------------
%    References
%----------------------------------------------------------------------------------------
\section*{References} % The \section*{} command stops section numbering

\addcontentsline{toc}{section}{\hspace*{-\tocsep}References} % Adds this section to the table of contents with negative horizontal space equal to the indent for the numbered sections
  \subsection*{This will be empty}  
    \paragraph {Use Papers2 for this part}
\blindtext


%----------------------------------------------------------------------------------------
%    Supplementary Data
%----------------------------------------------------------------------------------------

\end{document}