%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% Stylish Article
% LaTeX Template
% Version 1.0 (31/1/13)
%
% This template has been downloaded from:
% http://www.LaTeXTemplates.com
%
% Original author:
% Mathias Legrand (legrand.mathias@gmail.com)
%
% License:
% CC BY-NC-SA 3.0 (http://creativecommons.org/licenses/by-nc-sa/3.0/)
%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%----------------------------------------------------------------------------------------
%    PACKAGES AND OTHER DOCUMENT CONFIGURATIONS
%----------------------------------------------------------------------------------------

\documentclass[fleqn,10pt]{SelfArx} % Document font size and equations flushed left

\setlength{\columnsep}{0.55cm} % Distance between the two columns of text
\setlength{\fboxrule}{0.75pt} % Width of the border around the abstract

\definecolor{color1}{RGB}{0,100,0} % Color of the article title and sections
\definecolor{color2}{RGB}{238,238,0}% Color of the boxes behind the abstract and headings

\newlength{\tocsep} 
\setlength\tocsep{1.5pc} % Sets the indentation of the sections in the table of contents
\setcounter{tocdepth}{3} % Show only three levels in the table of contents section: sections, subsections and subsubsections

\usepackage{lipsum} % Required to insert dummy text
\usepackage[english]{babel} %more dumb text
\usepackage{blindtext}
\usepackage{pdfpages}

%----------------------------------------------------------------------------------------
%    ARTICLE INFORMATION
%----------------------------------------------------------------------------------------

\JournalInfo{BMC Cancer 2014, (IF = 14:189)\\
Stem Cells (7.7)\\ Stem Cell Research  (4.76)\\ 
European Journal of Cancer (IF = 5.0)\\ 
Genome Medicine (3.40) \\
Journal of Neuro-Oncology (3.12)\\ 
or similar journal} % Journal information

\Archive{Original article} % Additional notes (e.g. copyright, DOI, review/research article)

\PaperTitle{Functional and genome wide examination of the CD133 marker in Glioblastoma Multiforme Stem Cells} % Article title

\Authors{Brown, Daniel V\textsuperscript{1}*, Nicole Kountouri\textsuperscript{2}, D'Abaco, Giovanna\textsuperscript{2}, 
Ng, Wayne\textsuperscript{2}, Daniel, Paul M\textsuperscript{1}, Filiz, Gulay\textsuperscript{1}, Drummond, Kate\textsuperscript{2}
Lonie, Andrew\textsuperscript{3}, Morokoff, Andrew\textsuperscript{2}, Mantamadiotis, Theo\textsuperscript{1}} % Authors

\affiliation{\textsuperscript{1}\textit{Department of Pathology, University of Melbourne, Melbourne, Australia}} % Author affiliation
\affiliation{\textsuperscript{2}\textit{Department of Surgery, Royal Melbourne Hospital Melbourne, Australia}} % Author affiliation
\affiliation{\textsuperscript{3}\textit{Victorian Life Sciences Computational Initiative, Melbourne, Australia}} % Author affiliation
\affiliation{*\textbf{Corresponding author}: dvbrown@student.unimelb.edu.au} % Corresponding author

\Keywords{Cancer Stem Cells --- Glioblastoma --- CD133 --- Transcriptome} % Keywords - if you don't want any simply remove all the text between the curly brackets
\newcommand{\keywordname}{Keywords} % Defines the keywords heading name

%----------------------------------------------------------------------------------------
%	ABSTRACT
%----------------------------------------------------------------------------------------

\Abstract{
Glioblastoma multiforme (GBM) is an aggressive cancer of the brain with a median survival of 14 months. The cancer genome atlas was unable to resolve survival based on measurements from the tumour bulk. GBM has been shown to be maintained by a subpopulation of cells termed glioma-initiating cells (GICs). It may be these cells that are responsible for differences in survival.\\

The stem cell marker CD133 was used to isolate GICs by flow cytometry and the phenotypic properties of CD133+ and CD133- cells were compared by cellular and genome-wide methods. The transcriptome of a panel of human GIC samples was measured by RNA-seq to compare groups that differ by survival. The REST pathway was enriched in GICs obtained from short survivors and will be validated functionally.
}

\begin{document}
<<setup, include=FALSE, cache=FALSE>>=
library(knitr)
options(replace.assign=TRUE,width=50)
opts_chunk$set(fig.path='figure/graphics-', cache.path='cache/graphics-', fig.align='center', fig.width=5, fig.height=5, fig.show='hold', cache=TRUE, par=TRUE)
knit_hooks$set(par=function(before, options, envir){
if (before && options$fig.show!='none') par(mar=c(4,4,.1,.1),cex.lab=.95,cex.axis=.9,mgp=c(2,.7,0),tcl=-.3)
}, crop=hook_pdfcrop)
@
%\SweaveOpts{concordance=TRUE}
%This says that figures should be resized so that they fit the full width of the text.

\flushbottom % Makes all text pages the same height

\maketitle % Print the title and abstract box

\tableofcontents % Print the contents section

% \thispagestyle{empty} % Removes page numbering from the first page

%----------------------------------------------------------------------------------------
%    INTRODUCTION
%----------------------------------------------------------------------------------------
\section*{Introduction} % The \section*{} command stops section numbering

\addcontentsline{toc}{section}{\hspace*{-\tocsep}Introduction} % Adds this section to the table of contents with negative horizontal space equal to the indent for the numbered sections
  \subsection*{Some stats on prevalence and mortality}
  
Glioblastoma multiforme (GBM) is an aggressive, heterogeneous tumour of the central nervous system. These diverse morphological features have made diagnosis of GBM difficult (1). Molecular profiling of GBM tumours has revealed that similar to other tumours such as breast and ovarian cancer (2,3), GBM is not a single disease but consists of several distinct malignancies. Although GBM is a relatively rare type of cancer it has a 5 year survival of less than 5\% rendering it one of the most lethal types of tumours. Age is the most important risk factor for GBM, with incidence showing a 4th power relationship with age, indicative of a gradual accumulation of oncogenic mutations over time (7).

\subsection*{Some info on cancer stem cell theory}  
    \paragraph {Why CD133 is not a good marker}
GICs were initially characterised by expression of the stem cell marker CD133 (PROM1) and were able to form tumours from 100 cells in immune deficient NOD-SCID mice (41). 
CD133 or prominin-1 was originally described as being an epitope enriched in a hybridoma screen against CD34 isolated haemopoetic progenitors (42). 
The marker was later shown to be expressed by neural progenitor cells from human fetal brains (43). There are a number of limitations of the CD133 marker for glioma initiating cells. The protein itself is a 115-kDa membrane-associated glycoprotein with no known molecular function, conserved in humans, mice, worms and flies (44). 
The two major commercial antibodies target different glycoslyated regions of the protein, the precise locations of which are currently unknown. (45). CD133 undergoes mRNA splicing, generating at least 28 isoforms (46), contributing to the variation in detection by antibodies.\\
However CD133- cells are able to form tumours when 10 spheroids were orthotopically implanted into rat brains. The resulting tumours contained a small fraction of CD133+ cells (48).
Without a molecular marker to separate CD133- stem cells from non-stem cells it is difficult to examine the properties of this cell type. A variety of other stem cell markers have been trialed such as the adhesion molecules CD44, Integrin-alpha6, CD15 and the multiple-drug resistance protein ABCG2 but have not found wide spread use (45).\\
\textbf{Expand this intro section to discuss the other markers}

  \paragraph {Predictive significance of a stemness signature}
The clinical significance of CD133 and other stem cell markers are controversial. Immunohistochemistry does not show a statistically significant association of CD133 or CD15 positivity on survival (59). This is in contrast to studies at the mRNA level demonstrating that CD133 (PROM1) expression is a significant negative prognostic factor for both progression-free and overall survival in GBM (60).\\ 

  \paragraph {Phenotypes associated with GSCs}
Discuss contraversey and lack of reproducibility of some the phenotypic properties of GSCs.

\subsection*{Literature on studies which compare markers}  
    \paragraph {Studies that I am comparing myself to}
Pollard, Smith, Yan, Lottaz, Lopez etc.
    
%----------------------------------------------------------------------------------------
%    METHODS
%----------------------------------------------------------------------------------------
\section*{Methods} % The \section*{} command stops section numbering

\addcontentsline{toc}{section}{\hspace*{-\tocsep}Methods} % Adds this section to the table of contents with negative horizontal space equal to the indent for the numbered sections
  \subsection*{Cell Culture}  
    \paragraph {Culture of Glioma stem cells}
\blindtext

  \subsection*{Flow cytometry}  
    \paragraph {The important part of gating}
\blindtext vector of these gene names was then used to build the network of coexpressed genes using the manual module construction method.
    
  \subsection*{Functional assays}  
    \paragraph {Proliferation curve} a\\
    \paragraph {Limiting dilution (ELDA)} a\\
    \paragraph {Temozolomide sensitivity} a\\
    \paragraph {Invasion} a\\


  \subsection*{Library preparation and RNA sequencing}  
     \paragraph {RNA-seq libraries} a\\
\blindtext
    
  \subsection*{Bioinformatic analysis} 
    \paragraph {Preprocessing} a\\
    \paragraph {Differential expression testing} a\\
    \paragraph {Subtype classification} a\\
    \paragraph {Gene signature and survival analysis} a\\
\blindtext


%----------------------------------------------------------------------------------------
%    RESULTS
%----------------------------------------------------------------------------------------
\section*{RESULTS} % The \section*{} command stops section numbering

\addcontentsline{toc}{section}{\hspace*{-\tocsep}Results} % Adds this section to the table of contents with negative horizontal space equal to the indent for the numbered sections

  \subsection*{Isolation and characterisation of GSCs}
    \paragraph {Isolation and distribution of CD133 marker}
\blindtext

\begin{figure}[ht]
   \centering 
% Change this plot when you change the data source   
    \includegraphics[page=1, width=.75\linewidth]{/Users/d.brown6/Documents/facsData/140301_clone035_controlsAnnotatedV2.png}\\
    \includegraphics[page=1, width=.75\linewidth]{/Users/d.brown6/Documents/facsData/140301_differentPatients.png}
    \caption{ {How the GSCs were FACS sorted and the basis of the comparisons I use} }
    \label{fig:FACS}
\end{figure}

%------------------------------------------------------------------------------------------------------------------------

  \subsection *{Characterisation of the phenotypic properties of GSCs}
    \paragraph {The in vitro assays that compare CD133 - and CD133 +}
\blindtext

\begin{figure}[ht]
   \centering 
% Change this plot when you change the data source   
    \includegraphics[page=1, width=.95\linewidth]{/Users/d.brown6/Documents/Cell_biology/microscopy/ELDA/140417_elda_6clones/plots/mostMultiplePlot.pdf}
    \caption{{A slight but non-significant enrichment of sphere forming potential} }
    \label{fig:ELDA}
\end{figure}

%------------------------------------------------------------------------------------------------------------------------

\begin{figure}[ht]
   \centering 
% Change this plot when you change the data source   
    \includegraphics[page=1, width=.95\linewidth]{/Users/d.brown6/Documents/Cell_biology/proliferation/Resazurin/140417_6clones/analysis/plots/140430_cd133CollatedSummary.pdf}
    \caption{{CD133 cells grow slightly faster and are slightly more temozolomide sensitive} }
    \label{fig:Resazurin}
\end{figure}

%------------------------------------------------------------------------------------------------------------------------

\blindtext

\begin{figure}[ht]
   \centering 
% Change this plot when you change the data source   
   % \includegraphics[page=1, width=.75\linewidth]{/Users/d.brown6/Documents/Cell_biology/microscopy/invasion/140414_invasion/140417_day7/c12.tif}
    % \includegraphics[page=1, width=.75\linewidth]{/Users/d.brown6/Documents/Cell_biology/microscopy/invasion/140414_invasion/140417_day7/analysis/c12.tif}
    
    \includegraphics[page=1, width=.75\linewidth]{/Users/d.brown6/Documents/Cell_biology/microscopy/invasion/140414_invasion/plots/140430_summary.pdf}
    \includegraphics[page=1, width=.95\linewidth]{/Users/d.brown6/Documents/Cell_biology/microscopy/invasion/140414_invasion/plots/140430_invasionSummarisedCD133.pdf}

    \caption{{Not much consistency in invasion assay} }
    \label{fig:Resazurin}
\end{figure}


%------------------------------------------------------------------------------------------------------------------------

  \subsection *{Comparing the transcriptome of CD133 negative and positive cells}
    \paragraph {The in vitro assays that compare CD133 - and CD133 +}
\blindtext
    
\begin{figure}[ht]
   \centering 
% Change this plot when you change the data source  
    \includegraphics[page=1, width=.75\linewidth]{/Users/d.brown6/Documents/RNAdata/danBatch1/bowtieGem/revHTSeq/GLMedgeR/140203_facsBatch/140203_summaries.pdf}
    \includegraphics[page=1, width=.75\linewidth]{/Users/d.brown6/Documents/RNAdata/danBatch1/bowtieGem/revHTSeq/networks/131104_imageLegendBold.pdf}
    \includegraphics[page=1, width=.65\linewidth]{/Users/d.brown6/Documents/RNAdata/danBatch1/GSEA/original/131025_nonSignificantNatureGenes.pdf}
    
    \caption{{CD133 cells grow slightly faster and are slightly more temozolomide sensitive} }
    \label{fig:Resazurin}
\end{figure}

%------------------------------------------------------------------------------------------------------------------------

  
%------------------------------------------------------------------------------------------------------------------------



%----------------------------------------------------------------------------------------
%    DISCUSSION
%----------------------------------------------------------------------------------------
\section*{Discussion} % The \section*{} command stops section numbering

\addcontentsline{toc}{section}{\hspace*{-\tocsep}Discussion} % Adds this section to the table of contents with negative horizontal space equal to the indent for the numbered sections
  \subsection*{I will wildy speculate on my results here}  
    \paragraph {Say something clever here}
\lipsum


%----------------------------------------------------------------------------------------
%    ACKNOWLEDGEMENTS
%----------------------------------------------------------------------------------------
\section*{Acknowledgements} % The \section*{} command stops section numbering

\addcontentsline{toc}{section}{\hspace*{-\tocsep}Acknowledgements} % Adds this section to the table of contents with negative horizontal space equal to the indent for the numbered sections
    \paragraph {Shout out to}
Many thanks to Gulay Filiz for ordering my reagents, Nicole Kountouri and Wayne Ng for discussions about stem cells and Eric Joo for coffee runs. Also Colonel Sanders made a contribution to my nutrition. Prof Paul Waring for paying the bills.

%----------------------------------------------------------------------------------------
%    References
%----------------------------------------------------------------------------------------
\section*{References} % The \section*{} command stops section numbering

\addcontentsline{toc}{section}{\hspace*{-\tocsep}References} % Adds this section to the table of contents with negative horizontal space equal to the indent for the numbered sections
  \subsection*{This will be empty}  
    \paragraph {Use Papers2 for this part}
\blindtext


%----------------------------------------------------------------------------------------
%    Supplementary Data
%----------------------------------------------------------------------------------------

\end{document}