---
title: "Data Import"
description: |
  Import the dataset into a single-cell experiment
author: "Daniel Brown"
date: "`r Sys.Date()`"
output:
  html_document:
    toc: true
    toc_depth: 3
    toc_float:
      smooth_scroll: true
    theme: readable
    highlight: tango 
    df_print: paged
    code_folding: hide
editor_options: 
  chunk_output_type: console
---

```{r global_options, include=FALSE}
knitr::opts_chunk$set(fig.width=8, fig.asp=0.618, 
                      fig.path='/Figures/NNxx/DataImport/',
                      warning=FALSE, message=FALSE)
```

```{r setup, include=FALSE}
library(here)
library(tidyverse)
library(SingleCellExperiment)
library(Matrix)
library(AnnotationHub)
```

# Make single-cell experiment object

```{r}
counts <- matrix(rpois(100, lambda = 10), ncol=10, nrow=10)
samplesheet <- read.delim(file = "path to samplesheet")

# Construct a new SCE from the counts matrix
sce <- SingleCellExperiment(
  assays = list(counts = counts),
  colData=DataFrame(label=samplesheet),
    rowData=DataFrame(length=gene.info),
    metadata=list(plate="LCExxx")
  )
```

## Move ERCC to altexp slot

```{r}
is.spike <- grepl("^ERCC-", rownames(sce))
sce <- splitAltExps(sce, ifelse(is.spike, "ERCC", "gene"))
altExpNames(sce)
```

# Get chromosome names
```{r}
ah <- AnnotationHub()
query(ah, c("Homo sapiens", "Ensembl", "v97"))

# Annotate each gene with its chromosome location
ensdb <- ah[["key returned by query"]]
chromosome <- mapIds(ensdb, keys=rownames(sce),
    keytype="GENEID", column="SEQNAME")
rowData(sce)$chromosome <- chromosome
```

# Convert Ensembl gene ID to symbol
```{r}
rowData(sce)$ENSEMBL <- rownames(sce)
rowData(sce)$SYMBOL <- mapIds(ensdb, 
  keys=rownames(sce),
  keytype="GENEID", column="SYMBOL")
library(scater)
rownames(sce) <- uniquifyFeatureNames(rowData(sce)$ENSEMBL, 
  rowData(sce)$SYMBOL)
```

# Cleanup
```{r}
saveRDS(sce,
        here(
          "path_to_file"
        ))
sessionInfo()
```