---
title: "Quality Control"
description: |
  Remove low quality cells and genes not expressed
author: "Daniel Brown"
date: "`r Sys.Date()`"
output:
  html_document:
    toc: true
    toc_depth: 3
    toc_float:
      smooth_scroll: true
    theme: readable
    highlight: tango 
    df_print: paged
    code_folding: hide
editor_options: 
  chunk_output_type: console
---

```{r global_options, include=FALSE}
knitr::opts_chunk$set(fig.width=8, fig.asp=0.618, 
                      fig.path='/Figures/NNxx/QualityControl/',
                      warning=FALSE, message=FALSE)
```

```{r setup, include=FALSE}
library(here)
library(tidyverse)
library(scater)
library(scran)
library(DT)
```

# Read single-cell experiment object

```{r}
sce <- readRDS(here(
  "Path to sce"
))

chromosome <- rowData(sce)$chromosome
is.mito <- which(chromosome=="MT")
```

## Calculate quality control

```{r}
sce <- addPerCellQC(sce, 
  subsets=list(Mito=is.mito))

sce <- addPerFeatureQC(sce)
```

# Visualise QC metrics
```{r colData}
plotColData(sce, x="plate", y="detected")

plotColData(sce, x="plate", y="detected") +
    scale_y_log10()  + 
    facet_wrap(~phenotype)
```

# Remove low quality cells by adaptive threshold
```{r}
qc.lib <- isOutlier(sce$sum, log=TRUE, type="lower")
qc.nexprs <- isOutlier(sce$detected, log=TRUE,
  type="lower")
qc.spike <- isOutlier(sce$altexps_ERCC_percent,
  type="higher")
qc.mito <- isOutlier(sce$subsets_Mito_percent,
  type="higher")
discard <- qc.lib | qc.nexprs | qc.spike | qc.mito
```

# Summarize the number of cells removed for each reason.

`r attr(qc.lib, "thresholds")`  
`r attr(qc.nexprs, "thresholds")`  

```{r}
DataFrame(LibSize=sum(qc.lib), NExprs=sum(qc.nexprs),
  SpikeProp=sum(qc.spike), MitoProp=sum(qc.mito),
  Total=sum(discard))
```


# Cleanup
```{r}
saveRDS(sce,
        here(
          "path_to_file"
        ))
sessionInfo()
```