---
title: "Feature Selection"
description: |
  Choose the genes that are interesting for analysis
author: "Daniel Brown"
date: "`r Sys.Date()`"
output:
  html_document:
    toc: true
    toc_depth: 3
    toc_float:
      smooth_scroll: true
    theme: readable
    highlight: tango 
    df_print: paged
    code_folding: hide
editor_options: 
  chunk_output_type: console
---

```{r global_options, include=FALSE}
knitr::opts_chunk$set(fig.width=8, fig.asp=0.618, 
                      fig.path='/Figures/NNxx/FeatureSelection/',
                      warning=FALSE, message=FALSE)
```

```{r setup, include=FALSE}
library(here)
library(tidyverse)
library(scater)
library(scran)
library(DT)
```

# Read single-cell experiment object

```{r}
sce <- readRDS(here(
  "Path to sce"
))

chromosome <- rowData(sce)$chromosome
is.mito <- which(chromosome=="MT")
```

# Variance of the log-counts

```{r}
dec. <- modelGeneVar(sce.)
chosen <- getTopHVGs(dec., prop=0.1)
str(chosen)
```

```{r fit}
fit <- metadata()
plot(fit.$mean, fit.$var, 
  xlab="Mean of log-expression",
  ylab="Variance of log-expression")
curve(fit.$trend(x), col="dodgerblue",
  add=TRUE, lwd=2)
```

# Ordering by largest 'biological' variation 
Identify most interesting genes

```{r}
dec.[order(dec.$bio, decreasing=TRUE),]
```

# Cleanup
```{r}
saveRDS(sce,
        here(
          "path_to_file"
        ))
sessionInfo()
```