---
title: "Multi-sample comparisons"
author: "D_Brown"
date: "25/02/2020"
output: html_document
editor_options:
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Data import
```{r}
library(MouseGastrulationData)
sce.chimera <- WTChimeraData(samples=5:10)
```

# Feature-annotation

```{r}
library(scater)
rownames(sce.chimera) <- uniquifyFeatureNames(
    rowData(sce.chimera)$ENSEMBL, rowData(sce.chimera)$SYMBOL)
```

# quality-control
already performed by authors

```{r}
# We also remove cells labelled as stripped nuclei or doublets
drop <- sce.chimera$celltype.mapped %in% c("stripped", "Doublet")
sce.chimera <- sce.chimera[,!drop]
```

# normalization
using pre-computed size factors.

```{r}
sce.chimera <- logNormCounts(sce.chimera)
```

# variance-modelling
Retaining all genes with a positive biological component, to  preserve as much signal as possible in a very heterogeneous dataset.

```{r}
library(scran)
dec.chimera <- modelGeneVar(sce.chimera, block=sce.chimera$sample)
chosen.hvgs <- dec.chimera$bio > 0
```

# merging
hierarchical merge, first merge together reps with the # same genotype, then merge samples across different genotypes.

```{r}
library(batchelor)
set.seed(01001001)
merged <- correctExperiments(sce.chimera,
  batch=sce.chimera$sample,
  subset.row=chosen.hvgs,
  PARAM=FastMnnParam(
    merge.order=list(
      list(1,3,5), # WT (3 replicates)
      list(2,4,6)  # td-Tomato (3 replicates)
    )
  )
)
```

# clustering

```{r}
g <- buildSNNGraph(merged, use.dimred="corrected")
clusters <- igraph::cluster_louvain(g)
merged$cluster <- factor(clusters$membership)
```

# dimensionality-reduction
Using an external algorithm to compute nearest  neighbors for greater speed.  

```{r}
merged <- runTSNE(merged, dimred="corrected",
  external_neighbors=TRUE)
merged <- runUMAP(merged, dimred="corrected",
  external_neighbors=TRUE)

plotReducedDim(merged, colour_by = "celltype.mapped", dimred = "UMAP")
```

# Differential expression between conditions
## Creating pseudo-bulk samples

Using 'label' and 'sample' as our two factors; each column of the output corresponds to one unique combination of these two factors.

```{r}
summed <- aggregateAcrossCells(merged,
  id=DataFrame(
    label=merged$celltype.mapped,
    sample=merged$sample))
```

go from dim `r dim(merged)` of single-cells to  
dim `r dim(summed)` number of cells for psuesdo-bulking.  

Focus on cells labelled as 'Mesenchyme'.  
Need to convert to DGEList object for edgeR analysis.  

```{r}
label <- "Mesenchyme"
current <- summed[,label==summed$celltype.mapped]

# Construct a DGEList object for use in edgeR:
library(edgeR)
y <- DGEList(counts(current), samples=colData(current))
```

## Preprocessing
```{r}
# Discard sample-labels with very small library sizes
discarded <- isOutlier(y$samples$lib.size, log=TRUE,
  type="lower")
y <- y[,!discarded]

# Remove genes that are lowly expressed
keep <- filterByExpr(y, group=current$tomato)
y <- y[keep,]
```

## Compute normalization factors for the pseudo-bulk
these are different to the single-cell size factors
```{r}
y <- calcNormFactors(y)
```

## Statistical modelling
```{r}
# Construct design matrix, blocking on pool and
# including term for td-Tomato-status
design <- model.matrix(
  ~0 + factor(pool) + factor(tomato),
  y$samples)
# Tidy up design matrix column names
colnames(design) <- gsub(
  "factor|\\(|\\)|TRUE", "", colnames(design))

# Estimate and plot the negative binomial dispersions
y <- estimateDisp(y, design)
plotBCV(y)
```

### Estimate and plot the quasi-likelihood dispersions

```{r}
fit <- glmQLFit(y, design, robust=TRUE)
plotQLDisp(fit)
```

### Test for DE due to td-Tomato-status
This is what the coef="tomato" means.  

```{r}
res <- glmQLFTest(fit, coef="tomato")
# Summarise the results
summary(decideTests(res))
# View the top DE genes
topTags(res)
```


Pete says you can MNN correct multiple batches sequenctially.  
If sample 1 and 2 are more heterogenous MNN them first then merge on sample 3.  
You want to start with the most number of cell types in the first merge.  
The MNN package can assist in deciding which order to integrate on.  

Can use marker genes instead of variable genes in the analysis if there is heterogeneity you want to focus on.  

# Perform DE analysis on Cardiomyocytes cell type
```{r}
label <- "Cardiomyocytes"
current <- summed[,label==summed$celltype.mapped]

# Construct a DGEList object for use in edgeR:
library(edgeR)
y <- DGEList(counts(current), samples=colData(current))

# Discard sample-labels with very small library sizes
discarded <- isOutlier(y$samples$lib.size, log=TRUE,
  type="lower")
y <- y[,!discarded]

# Remove genes that are lowly expressed
keep <- filterByExpr(y, group=current$tomato)
y <- y[keep,]

y <- calcNormFactors(y)
```

```{r}
design <- model.matrix(
  ~0 + factor(pool) + factor(tomato),
  y$samples)
# Tidy up design matrix column names
colnames(design) <- gsub(
  "factor|\\(|\\)|TRUE", "", colnames(design))

y <- estimateDisp(y, design)
plotBCV(y)

# Estimate and plot the quasi-likelihood dispersions
fit <- glmQLFit(y, design, robust=TRUE)
plotQLDisp(fit)

# Test for DE due to td-Tomato-status
res <- glmQLFTest(fit, coef="tomato")
# Summarise the results
summary(decideTests(res))
# View the top DE genes
topTags(res)
```

# Loop over all cell types for DE
There is some very fancy loops and try except clauses in here.  

```{r}
de.results <- list()

for (i in unique(summed$celltype.mapped)) {
  current <- summed[,i==summed$celltype.mapped]
  y <- DGEList(counts(current), samples=colData(current))

  discarded <- isOutlier(
    colSums(counts(current)), log=TRUE, type="lower")
  y <- y[,!discarded]
  y <- y[filterByExpr(y, group=current$tomato),]

  y <- calcNormFactors(y)

  design <- try(
    model.matrix(
      ~0 + factor(pool) + factor(tomato), y$samples),
    silent=TRUE)
    if (is(design, "try-error") ||
      qr(design)$rank==nrow(design) ||
      qr(design)$rank < ncol(design)) {
        next
    }
  colnames(design) <- gsub(
    "factor|\\(|\\)|TRUE", "", colnames(design))

    y <- estimateDisp(y, design)
    fit <- glmQLFit(y, design)
    res <- glmQLFTest(fit, coef="tomato")
    de.results[[i]] <- res
}

```

## Tabulate DEGs per label
```{r}
library(DT)
summaries <- lapply(
  de.results,
  FUN=function(x) summary(decideTests(x))[,1])
sum.tab <- do.call(rbind, summaries)
datatable(sum.tab)
```

## Find genes consistently DE across labels
More fancy code that unlists multiple results and sorts them into a flat object.  

```{r}
degs <- lapply(
  de.results,
  FUN=function(x) rownames(topTags(x, p.value=0.05)))
common.degs <- sort(table(unlist(degs)),
  decreasing=TRUE)

common.degs
```
